Sources/PackageModel/ManifestSourceGeneration.swift

```diff
@@ -100,22 +100,32 @@ fileprivate extension SourceCodeFragment {
     
     /// Instantiates a SourceCodeFragment to represent a single platform.
     init(from platform: PlatformDescription) {
-        // NOTE: This could be cleaned up to use the nicer version accessors.
-        switch platform.platformName {
-        case "macos":
-            self.init(enum: "macOS", string: platform.version)
-        case "maccatalyst":
-            self.init(enum: "macCatalyst", string: platform.version)
-        case "ios":
-            self.init(enum: "iOS", string: platform.version)
-        case "tvos":
-            self.init(enum: "tvOS", string: platform.version)
-        case "watchos":
-            self.init(enum: "watchOS", string: platform.version)
-        case "driverkit":
-            self.init(enum: "DriverKit", string: platform.version)
-        default:
-            self.init(enum: platform.platformName, string: platform.version)
+        // Provides knowledge about what the current version of PackageDescription provides.
+        struct PlatformManifestRepresentation {
+            var manifestName: String
+            var knownVersions: Set<String>
+        }
+        let platformManifestReps = [
+            "macos": PlatformManifestRepresentation(manifestName: "macOS", knownVersions: ["10.10", "10.11", "10.12", "10.13", "10.14", "10.15", "11", "12"]),
```

Comment:
nit: can use `.init` instead of long form if the array type is defined

Reply:
I actually had it that way to start with but I dislike having to add a type to the collection more than I do spelling out the type.  But I can change it, I don't feel superstrongly.